### PR TITLE
Update CONTRIBUTING.md file.

### PR DESCRIPTION
**Pull request by sile**
This PR contains the following changes to `CONTRIBUTING.md`:
- Added a link to [Coding Style Conventions](https://github.com/pfnet/optuna/wiki/Coding-Style-Conventions) wiki page.
- Stopped requesting contributors to run CI jobs in their local environment because it's too time and network consuming.

### CODE REVIEW

**Comment by sile** [comment]
I merged your suggestions. Thank you! > @hvy

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -80,4 +82,4 @@ if it is rendered as expected.
 
 Note that docstrings (documentation comments in the source code) are collected from the installed
 Optuna module. If you modified docstrings, make sure to install the module (e.g.,
-using `pip install -e .`) before building the documentation.
\ No newline at end of file
+using `pip install -e .`) before building the documentation.
```
Comment (hvy) [inline CONTRIBUTING.md]: What's the diff on this line?

**Comment by hvy** [comment]
Thanks for the quick response, LGTM!

File: CONTRIBUTING.md (hunk 1 above)
Comment (sile) [reply CONTRIBUTING.md]: I think that my editor automatically added a newline character to the end of this line.
I can remove this diff if you're concerned.

Comment (hvy) [reply CONTRIBUTING.md]: Ah no worries, I was just curious since it wasn't rendered 🙂

**Pull request merged**